File: testsprite-tests/be-distributions.py
import requests

TARGET_URL = "https://yemen-telecom.onrender.com"

def get_manager_token():
    r = requests.post(f"{TARGET_URL}/api/auth/login", json={"username": "manager", "password": "Admin@123"}, timeout=30)
    return r.json()["token"]

def auth_header(token):
    return {"Authorization": f"Bearer {token}"}

def test_list_distributions():
    token = get_manager_token()
    r = requests.get(f"{TARGET_URL}/api/distributions", headers=auth_header(token), timeout=30)
    assert r.status_code == 200, f"Expected 200, got {r.status_code}"

def test_pending_count():
    token = get_manager_token()
    r = requests.get(f"{TARGET_URL}/api/distributions/pending-count", headers=auth_header(token), timeout=30)
    assert r.status_code == 200, f"Expected 200, got {r.status_code}"
    assert "count" in r.json()

def test_create_distribution_as_manager_forbidden():
    token = get_manager_token()
    r = requests.post(f"{TARGET_URL}/api/distributions", json={"operator": "Yemen Mobile", "count": 10}, headers=auth_header(token), timeout=30)
    assert r.status_code == 403, f"Expected 403, got {r.status_code}"

test_list_distributions()
test_pending_count()
test_create_distribution_as_manager_forbidden()

File: testsprite-tests/be-distributions.js
const BASE = 'https://yemen-telecom.onrender.com';

async function request(path, options = {}) {
  const res = await fetch(`${BASE}${path}`, {
    headers: { 'Content-Type': 'application/json', ...options.headers },
    ...options,
  });
  const body = res.headers.get('content-type')?.includes('json') ? await res.json() : await res.text();
  return { status: res.status, body, headers: Object.fromEntries(res.headers) };
}

async function test(name, fn) {
  try {
    await fn();
    console.log(`PASS: ${name}`);
  } catch (e) {
    console.error(`FAIL: ${name} — ${e.message}`);
    process.exitCode = 1;
  }
}

function assert(condition, msg) {
  if (!condition) throw new Error(msg || 'Assertion failed');
}

let managerToken;

async function loginManager() {
  const { body } = await request('/api/auth/login', {
    method: 'POST',
    body: JSON.stringify({ username: 'manager', password: 'Admin@123' }),
  });
  managerToken = body.token;
}

async function run() {
  await loginManager();
  if (!managerToken) {
    console.error('Cannot run Distribution tests: manager login failed');
    process.exitCode = 1;
    return;
  }

  const auth = { Authorization: `Bearer ${managerToken}` };

  await test('GET /api/distributions returns array', async () => {
    const { status, body } = await request('/api/distributions', { headers: auth });
    assert(status === 200, `Expected 200, got ${status}`);
    assert(Array.isArray(body) || Array.isArray(body.data), 'Expected array');
  });

  await test('GET /api/distributions/pending-count returns count', async () => {
    const { status, body } = await request('/api/distributions/pending-count', { headers: auth });
    assert(status === 200, `Expected 200, got ${status}`);
    assert(typeof body.count === 'number', 'Expected count number');
  });

  await test('POST /api/distributions as manager returns 403', async () => {
    const { status } = await request('/api/distributions', {
      method: 'POST',
      headers: auth,
      body: JSON.stringify({
        operator: 'Yemen Mobile',
        count: 10,
      }),
    });
    assert(status === 403, `Expected 403, got ${status}`);
  });
}

run();
